mantle/cmd/kola/qemuexec.go

```diff
@@ -147,20 +147,24 @@ func runQemuExec(cmd *cobra.Command, args []string) error {
 	if directIgnition && ignition == "" {
 		return fmt.Errorf("Cannot use ignition-direct without a path to an Ignition config")
 	}
-	if !directIgnition {
-		if ignition == "" {
+	if !directIgnition && ignition != "" {
+		buf, err := ioutil.ReadFile(ignition)
+		if err != nil {
+			return err
+		}
+		config, err = conf.Ignition(string(buf)).Render(kola.IsIgnitionV2())
+		if err != nil {
+			return errors.Wrapf(err, "parsing %s", ignition)
+		}
+	}
+
+	ensureConfig := func() {
```

Comment:
Hard not to see the Rust `let mut config = Option<Config>; .. config.get_or_insert_with()` here.